### PR TITLE
boost: 1.65.0 Update + Fixes

### PR DESCRIPTION
**Pull request by ClaymorePT**
Maintainer: @ClaymorePT
Compile tested:  Raspberry Pi 3 (BCM2710)
Run tested: None

Description:
This commit updates the boost libraries to version 1.65.0

This new version brings two new libraries:
 - Boost.Stacktrace [1]
   -> Gather, store, copy and print backtraces, from Antony Polukhin.
 - Boost.PolyCollection [2]
   -> Fast containers of polymorphic objects, from Joaquín M López Muñoz.

Fixes in this commit:
 - Boost.Python for Python 3.6
   -> The dynamic library was not being compiled, resulting in an empty package. Fixed.
 - Boost.TypeErasure
   -> The dynamic library was not available to install. Fixed.
 - Boost.Serialization
   -> The wserialization dynamic library was being compiled but not available to install. Fixed.

Extras:
 - This commit also contains the post-release fixes, which did not make it into the final release [3].

As always, more information about this release, can be found in [4].

[1]: http://www.boost.org/doc/libs/1_65_0/doc/html/stacktrace.html
[2]: http://www.boost.org/doc/libs/1_65_0/doc/html/poly_collection.html
[3]: http://www.boost.org/patches/
[4]: http://www.boost.org/users/history/version_1_65_0.html

Signed-off-by: Carlos Miguel Ferreira <carlosmf.pt@gmail.com>

### CODE REVIEW

File: libs/boost/Makefile [hunk 1]
```diff
@@ -16,15 +16,15 @@ include $(INCLUDE_DIR)/nls.mk
...
+PKG_VERSION:=1.65.0
+PKG_SOURCE_VERSION:=1_65_0
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)_$(PKG_SOURCE_VERSION).tar.bz2
 PKG_SOURCE_URL:=https://sourceforge.net/projects/boost/files/boost/$(PKG_VERSION)
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)_$(PKG_SOURCE_VERSION)
 HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/$(PKG_NAME)_$(PKG_SOURCE_VERSION)
-PKG_MD5SUM:=7bcc5caace97baa948931d712ea5f37038dbb1c5d89b43ad4def4ed7cb683332
+PKG_MD5SUM:=ea26712742e2fb079c2a566a31f3266973b76e38222b9f88b387e3c8b2f9902c
```
Comment (thess) [inline libs/boost/Makefile]: According to Travis CI
```WARNING: Use of PKG_MD5SUM is deprecated, switch to PKG_HASH```

Comment (ClaymorePT) [reply libs/boost/Makefile]: @thess 
Thanks!
One question. Should I pay attention to the Travis failure? Is it working properly?

Comment (champtar) [reply libs/boost/Makefile]: Hi @ClaymorePT 
The only "normal" failure is when travis timeout (we only have 2 vcpus for 60 mins, sometimes it's not enough to rebuild all dependencies)
Else everytime Travis is not working properly don't hesitate to ping me

Comment (ClaymorePT) [reply libs/boost/Makefile]: @champtar 

Ok. Thank you for the warning. For now, it is still building the package. I will wait.

**Comment by champtar** [comment]
Hi @ClaymorePT 
Just pushed a conflicting commit, can you rebase?

**Comment by ClaymorePT** [comment]
@champtar 
Travis blew up

> Pull request should not include merge commits[m
> Author name (Carlos Miguel Ferreira) seems ok
> Commit subject line MUST start with '<package name>: ' (Merge branch 'master' into boost_1.65.0_r1)
> Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: Carlos Miguel Ferreira <ClaymorePT@users.noreply.github.com>')

I'll just close this push request and open a new one, since I made the mistake of updating my repo, inside the branch I'm suppost to push.

**Pull request closed without merging**